pkg/common/patterns.go

```diff
@@ -58,14 +58,14 @@ func (r RegexState) Matches(data []byte) []string {
 
 // UsernameRegexCheck constructs an username usernameRegex pattern from a given pattern of excluded characters.
 func UsernameRegexCheck(pattern string) RegexState {
-	raw := fmt.Sprintf(`(?im)(?:user|usr)\S{0,40}?[:=\s]{1,3}[ '"=]{0,1}([^:%+v]{4,40})\b`, pattern)
+	raw := fmt.Sprintf(`(?im)(?:user|usr)\S{0,40}?[:=\s]{1,3}[ '"=]{0,1}([^:%+v'"\s]{4,40})`, pattern)
 
 	return RegexState{regexp.MustCompile(raw)}
 }
 
-// PasswordRegexCheck constructs an username usernameRegex pattern from a given pattern of excluded characters.
+// PasswordRegexCheck constructs a password passwordRegex pattern from a given pattern of excluded characters.
 func PasswordRegexCheck(pattern string) RegexState {
-	raw := fmt.Sprintf(`(?im)(?:pass|password)\S{0,40}?[:=\s]{1,3}[ '"=]{0,1}([^:%+v]{4,40})`, pattern)
+	raw := fmt.Sprintf(`(?im)(?:pass)\S{0,40}?[:=\s]{1,3}[ '"=]{0,1}([^:%+v'"\s]{4,40})`, pattern)
```

Comment:
@dylanTruffle any reason for adding `'` and `"` to the excluded chars in password? for username that makes sense, but for password those are valid - no?

Reply:
I'm guessing he was adding that for some boundary condition, but there are better ways to do that. Right now it doesn't look like there is any sort of boundary.

Reply:
makes sense. i still have concerns that `'` or `"` may be valid chars. 

for example, Couchbase, the detector for which uses this library, allows both as valid special characters: [couchbase password policy](https://docs.couchbase.com/server/current/rest-api/rest-set-password-policy.html#curl-syntax:~:text=Acceptable%20special%20characters)

perhaps we need to add additional mechanisms to catch or filter passwords within quotes

Reply:
In my experience this can't be done in a satisfying way without defining the pattern *n* times. Otherwise it is not exhaustive and is prone to errors.

E.g.,
```regexp
\s([^\s]+)\s|"([^"]+)"|'([^']+)'
```